src/tests/ftest/datamover/copy_negative.py

```diff
@@ -0,0 +1,478 @@
+
+    def test_copy_error_check(self):
+        """Jira ID: DAOS-5515
+        Test Description:
+            (1) Error checking: destination filename is invalid.
+            (2) Error checking: destination pool out of space.
+            (3) Error checking: destination POSIX file system out of space.
+        :avocado: tags=all,datamover
+        :avocado: tags=copy_negative,copy_error_check
+        """
+        # Create pool and containers
+        pool = self.create_pool()
+        container1 = self.create_cont(pool)
+
+        # (1) Destination filename is invalid
+        # Create test file
+        self.write_daos(pool, container1)
+        # Use a really long filename
+        dst_filename = "d"*300
+        dst = os.path.join(self.posix_test_path, dst_filename)
+        self.run_dcp(
+            source=self.daos_test_file, target=dst,
+            src_pool=pool, src_cont=container1,
+            test_desc="error_check (1)",
+            expected_error_code=self.MFU_ERR_DCP_COPY,
+            expected_errno=36) # File name too long
+
+        # (2) Destination pool out of space.
+        # Write a large file to POSIX
+        self.ior_cmd.block_size.update(self.block_size_large)
+        self.write_posix()
+        self.run_dcp(
+            source=self.posix_test_file, target=self.daos_test_file,
+            dst_pool=pool, dst_cont=container1,
+            test_desc="error_check (2)",
+            expected_error_code=self.MFU_ERR_DCP_COPY,
+            expected_errno=28) # No space left on device
+
+        # TODO: how to limit the space?
+        # (3) Destination POSIX file system out of space.
+
+    def write_daos(self, pool, container):
+        """Uses ior to write the test file to a DAOS container."""
+        self.ior_cmd.api.update("DFS")
+        self.ior_cmd.flags.update(self.flags_write)
+        self.ior_cmd.test_file.update(self.daos_test_file)
+        self.ior_cmd.set_daos_params(self.server_group, pool, container.uuid)
+        out = self.run_ior(self.get_ior_job_manager_command(), self.processes)
+
+    def write_posix(self, test_file=None):
+        """Uses ior to write the test file in POSIX."""
+        self.ior_cmd.api.update("POSIX")
+        self.ior_cmd.flags.update(self.flags_write)
+        if test_file is None:
+            self.ior_cmd.test_file.update(self.posix_test_file)
+        else:
+            self.ior_cmd.test_file.update(test_file)
+        self.ior_cmd.set_daos_params(self.server_group, self.pool)
+        out = self.run_ior(self.get_ior_job_manager_command(), self.processes)
+
+    def nvl(self, s, null_val=None):
+        """ Returns null_val if s is None or empty.
+            Else, returns s"""
+        if s is None:
+            return null_val
+        if not str(s):
+            return null_val
+        return s
+
+    def extract_rc(self, s):
+        """Extracts the rc from a CommandFailure error"""
+        rc_search = re.search("\(rc=([0-9]+)\)", str(s))
```

Comment:
(lint) '\('. String constant might be missing an r prefix.
(lint) '\)'. String constant might be missing an r prefix.